src/Vault.sol

```diff
@@ -0,0 +1,398 @@
+
+    function getAssets() public view returns (address[] memory) {
+        return _getAssetStorage().list;
+    }
+
+    function getAsset(address asset_) public view returns (AssetParams memory) {
+        return _getAssetStorage().assets[asset_];
+    }
+
+    function getStrategies() public view returns (address[] memory) {
+        return _getStrategyStorage().list;
+    }
+
+    function getStrategy(address asset_) public view returns (StrategyParams memory) {
+        return _getStrategyStorage().strategies[asset_];
+    }
+
+    function previewDepositAsset(address asset_, uint256 assets_) public view returns (uint256) {
+        if (!_getAssetStorage().assets[asset_].active) revert InvalidAsset();
+        return _convertToShares(asset_, assets_, Math.Rounding.Floor);
+    }
+
+    function depositAsset(address asset_, uint256 assets_, address receiver) public nonReentrant returns (uint256) {
+        if (paused()) revert Paused();
+        if (!_getAssetStorage().assets[asset_].active) revert InvalidAsset();
+
+        uint256 shares = previewDepositAsset(asset_, assets_);
+        _deposit(asset_, _msgSender(), receiver, assets_, shares);
+
+        return shares;
+    }
+
+    function paused() public view returns (bool) {
+        return _getVaultStorage().paused;
+    }
+
+    function rateProvider() public view returns (address) {
+        return _getVaultStorage().rateProvider;
+    }
+
+    function bufferStrategy() public view returns (address) {
+        return _getVaultStorage().bufferStrategy;
+    }
+
+    function processAccounting() public {
+        AssetStorage storage assetStorage = _getAssetStorage();
+        StrategyStorage storage strategyStorage = _getStrategyStorage();
+        uint256 totalBaseBalance = 0;
+
+        for (uint256 i = 0; i < assetStorage.list.length; i++) {
+            address asset_ = assetStorage.list[i];
+            uint256 idleBalance = IERC20(asset_).balanceOf(address(this));
+            assetStorage.assets[asset_].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(asset_, idleBalance);
+        }
+
+        for (uint256 i = 0; i < strategyStorage.list.length; i++) {
+            address strategy = strategyStorage.list[i];
+            uint256 idleBalance = IERC20(strategy).balanceOf(address(this));
+            strategyStorage.strategies[strategy].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(strategy, idleBalance);
+        }
+
+        _getVaultStorage().totalAssets = totalBaseBalance;
+    }
+
+    function processAllocation(address[] calldata targets, uint256[] memory values, bytes[] calldata data)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        for (uint256 i = 0; i < targets.length; i++) {
+            bool success;
+            bytes memory returnData;
+
+            // Determine the type of target and process accordingly
+            if (_getAssetStorage().assets[targets[i]].active) {
```

Comment:
the active boolean is used to guard both deposit calls and processAllocation.

We may want to turn off an asset but still manage it with processAllocation.

Not sure it's needed here at all,  and if the flag is needed it should be decoupled from the one guarding deposits

Reply:
I've refactored this processor function, interested to get your feedback
https://github.com/yieldnest/yieldnest-vault/blob/0ec0acf6d968612a3611127f67017d9c976d2306/src/Vault.sol#L219